mlflow/server/otel_api.py

```diff
@@ -0,0 +1,100 @@
+"""
+OpenTelemetry REST API endpoints for MLflow FastAPI server.
+
+This module implements the OpenTelemetry Protocol (OTLP) REST API for ingesting spans
+according to the OTel specification:
+https://opentelemetry.io/docs/specs/otlp/#otlphttp
+
+Note: This is a minimal implementation that serves as a placeholder for the OTel endpoint.
+The actual span ingestion logic would need to properly convert incoming OTel format spans
+to MLflow spans, which requires more complex conversion logic.
+"""
+
+from typing import Any
+
+from fastapi import APIRouter, Header, HTTPException, Request, status
+from google.protobuf.message import DecodeError
+from opentelemetry.proto.collector.trace.v1.trace_service_pb2 import ExportTraceServiceRequest
+from pydantic import BaseModel, Field
+
+from mlflow.entities.span import Span
+from mlflow.server.handlers import _get_tracking_store
+from mlflow.tracing.utils.otlp import MLFLOW_EXPERIMENT_ID_HEADER, OTLP_TRACES_PATH
+
+# Create FastAPI router for OTel endpoints
+otel_router = APIRouter(prefix=OTLP_TRACES_PATH, tags=["OpenTelemetry"])
+
+
+class OTelExportTraceServiceResponse(BaseModel):
+    """
+    Pydantic model for the OTLP/HTTP ExportTraceServiceResponse.
+
+    This matches the OpenTelemetry protocol specification for trace export responses.
+    """
+
+    partialSuccess: dict[str, Any] | None = Field(
+        None, description="Details about partial success of the export operation"
+    )
+
+
+@otel_router.post("", response_model=OTelExportTraceServiceResponse, status_code=200)
+async def export_traces(
+    request: Request,
+    x_mlflow_experiment_id: str = Header(..., alias=MLFLOW_EXPERIMENT_ID_HEADER),
+) -> OTelExportTraceServiceResponse:
+    """
+    Export trace spans to MLflow via the OpenTelemetry protocol.
+
+    This endpoint accepts OTLP/HTTP protobuf trace export requests.
+
+    Args:
+        request: OTel ExportTraceServiceRequest in protobuf format
+        x_mlflow_experiment_id: Required header containing the experiment ID
+
+    Returns:
+        OTel ExportTraceServiceResponse indicating success
+
+    Raises:
+        HTTPException: If the request is invalid or span logging fails
+    """
+    body = await request.body()
```

Comment:
I'm not sure if this is necessary, but should we validate that the content type is `application/x-protobuf`?

Reply:
Added!